RF24.cpp

```diff
@@ -770,19 +801,175 @@ void RF24::printPrettyDetails(void) {
+                                   "\n\n"
+                                  );
+
+
+  char tx_address_char_array[16] = {'\0'};
+  sprintf_address_register(tx_address_char_array, TX_ADDR);
+
+  char pipe_address_char_2d_array[6][16] = {'\0'};
+  char pipe_address_char_array[16] = {'\0'};
+  bool isOpen_array[6] = {false};
+
+  uint8_t openPipes = read_register(EN_RXADDR);
+  for (uint8_t i = 0; i < 6; ++i)
+  {
+    bool isOpen = openPipes & _BV(i);
+    isOpen_array[i] = isOpen;
+    if (i < 2)
+    {
+      sprintf_address_register(pipe_address_char_array, static_cast<uint8_t>(RX_ADDR_P0 + i));
+      for (uint8_t j = 0; j < 16; j++)
+      {
+        pipe_address_char_2d_array[i][j] = pipe_address_char_array[j];
+      }
+      for (uint8_t j = 0; j < 16; j++)
+      {
+        pipe_address_char_array[j] = '\0';
+      }
+    }
+    else
+    {
+      sprintf_byte_register(pipe_address_char_array, static_cast<uint8_t>(RX_ADDR_P0 + i));
+      for (uint8_t j = 0; j < 16; j++)
+      {
+        pipe_address_char_2d_array[i][j] = pipe_address_char_array[j];
+      }
+      for (uint8_t j = 0; j < 16; j++)
+      {
+        pipe_address_char_array[j] = '\0';
+      }
+    }
+  }
+
+  sprintf_P(debugging_information,
+            format_string,
+            csn_pin,
+            ce_pin,
+            static_cast<uint8_t>(spi_speed / 1000000),
+            getChannel(),
+            static_cast<uint16_t>(getChannel() + 2400),
+            (char *)(pgm_read_ptr(&rf24_datarate_e_str_P[getDataRate()])),
+            (char *)(pgm_read_ptr(&rf24_pa_dbm_e_str_P[getPALevel()])),
+            (char *)(pgm_read_ptr(&rf24_feature_e_str_P[(read_register(RF_SETUP) & 1) * 1])),
+            (char *)(pgm_read_ptr(&rf24_crclength_e_str_P[getCRCLength()])),
+            ((read_register(SETUP_AW) & 3) + 2),
+            getPayloadSize(),
+            ((read_register(SETUP_RETR) >> ARD) * 250 + 250),
+            (read_register(SETUP_RETR) & 0x0F),
+            (read_register(OBSERVE_TX) >> 4),
+            (read_register(OBSERVE_TX) & 0x0F),
+            (char *)(pgm_read_ptr(&rf24_feature_e_str_P[static_cast<bool>(read_register(FEATURE) & _BV(EN_DYN_ACK)) * 2])),
+            (char *)(pgm_read_ptr(&rf24_feature_e_str_P[static_cast<bool>(read_register(FEATURE) & _BV(EN_ACK_PAY)) * 1])),
+            (char *)(pgm_read_ptr(&rf24_feature_e_str_P[(read_register(DYNPD) && (read_register(FEATURE) &_BV(EN_DPL))) * 1])),
+            static_cast<char>(static_cast<bool>(read_register(EN_AA) & _BV(ENAA_P5)) + 48),
+            static_cast<char>(static_cast<bool>(read_register(EN_AA) & _BV(ENAA_P4)) + 48),
+            static_cast<char>(static_cast<bool>(read_register(EN_AA) & _BV(ENAA_P3)) + 48),
+            static_cast<char>(static_cast<bool>(read_register(EN_AA) & _BV(ENAA_P2)) + 48),
+            static_cast<char>(static_cast<bool>(read_register(EN_AA) & _BV(ENAA_P1)) + 48),
+            static_cast<char>(static_cast<bool>(read_register(EN_AA) & _BV(ENAA_P0)) + 48),
```

Comment:
I assume this means that the auto-ack feature is always represented in binary format? This is another difference that should be noted in the docs for this function.

Originally, it should display `true`/`false` when all the pipes are uniformly configured.

Reply:
I reenabled that behavior, the pointer points to the string literal "Enabled" if true else it prints out bitwise which I believe is the default behavior.